src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1845,8 +1847,12 @@ private function registerMessengerConfiguration(array $config, ContainerBuilder
                 throw new LogicException(sprintf('Invalid Messenger configuration: the failure transport "%s" is not a valid transport or service id.', $config['failure_transport']));
             }
 
+            $transports = array_keys($config['transports']);
+            $numberOfTransports = count($transports);
+            $failedTransports = array_combine($transports, array_fill(0, $numberOfTransports, $senderReferences[$config['failure_transport']]));
+
             $container->getDefinition('messenger.failure.send_failed_message_to_failure_transport_listener')
-                ->replaceArgument(0, $senderReferences[$config['failure_transport']]);
+                ->replaceArgument(0, $failedTransports);
             $container->getDefinition('console.command.messenger_failed_messages_retry')
```

Comment:
what can we do with the commands for the retries if we have now multiple failed transports.
Should I create one command per transport with a failed queue defined?